### ISSUE TITLE
IMPORTANT: Looking for contributors

### ISSUE THREAD

**Issue by domibarton** [confirm/PhpZabbixApi#28]
Hi there

_TL;DR: Shifted from PHP to Python, no longer using PHP, looking for contributors._

A long time ago I've had a customer who was looking for a shiny & management-compliant monitoring dashboard. Because all the data was already available in Zabbix, I started to create a PHP-based WebUI which was accessing the Zabbix API. Unfortunately, I couldn't find a nice PHP API at that time and I started to develop my own API. This was when PhpZabbixApi was born!

A while ago I started to develop one of my projects in [Python](https://www.python.org/) and after about one year my shift from PHP to Python (and [Django](https://www.djangoproject.com/)) was complete. I've never looked back and I'm quite happy and impressed by Python and Django (lovely lovely Python & Django)!

Unfortunately, this means I'm no longer the "optimal" candidate to maintain my own code (don't know if I ever was ^^). This is one of my last PHP projects and it's kinda hard to support it today (my PHP knowledge is getting worse every day). I'm also relatively busy lately and therefore some of you already noticed that my response time is quite bad (sorry guys).

Long story short: I'm looking for contributors which help me to support the API, so that the community finally gets the support they deserve. Just send me a mail or comment this issue if you're interested in contributing - that would be awesome!

Cheers
Domi

**Comment by chadmandoo** [comment]
Hey I have a client implementing the Zabbix API and I will be using your library. I will be more than willing to help support the API.

**Comment by domibarton** [comment]
Hi @chadmandoo 

Unfortunately, I didn't see your message… Sorry for that.
Anyway, thanks for reaching out - this is great to hear :)

I'll add you as a contributor, just give me a second.

Cheers & Thanks
Domi

**Comment by domibarton** [comment]
@chadmandoo created a new team and added you to it…

**Comment by phansys** [comment]
Hi @domibarton, I'm interested in the development of this library, so I'm offering myself as contributor.
Many thanks for your work.

**Comment by ojgarciab** [comment]
Hi @domibarton I am using zabbix 2.2 and I'm migrating to 4.0, all my scripts based on PhpZabbixApi stopped working and composer still in 2.4 version. I'd like to help fixing/updating code.

**Comment by phansys** [comment]
@ojgarciab, if you have a chance, I'd like to check with you if the changes proposed at #39 fits with your needs. Please let me know and I'll try to provide any support that you could require.
Thank you in advance.

**Comment by ojgarciab** [comment]
> @ojgarciab, if you have a chance, I'd like to check with you if the changes proposed at #39 fits with your needs. Please let me know and I'll try to provide any support that you could require.

Great work (and a huge merge): I'll wait for the code to be mixed to start helping with improvements.

The changes I made to my code was (from working 2.2):

```
{
    "alias": "ALIAS",
    "name": "NAME",
    "surname": "SURNAME",
    "passwd": "LDAP",
    "usrgrps": [
        1,
        2
    ],
    "user_medias": [
        {
            "mediatypeid": "1",
            "sendto": "e@mail",
            "active": 0,
            "severity": 63,
            "period": "1-7,00:00-24:00"
        }
    ]
}
```

To (zabbix 4.0):

```
{
    "alias": "ALIAS",
    "name": "NAME",
    "surname": "SURNAME",
    "passwd": "LDAP",
    "usrgrps": [
        {
            "usrgrpid": 1
        },
        {
            "usrgrpid": 2
        }
    ],
    "user_medias": [
        {
            "mediatypeid": "1",
            "sendto": [
                "e@mail"
            ],
            "active": 0,
            "severity": 63,
            "period": "1-7,00:00-24:00"
        }
    ]
}
```

The API error messages are very confusing:
> API error -32602: Invalid parameter "/1/usrgrps/1": an array is expected.

So I'd try to improve type restrictions at library side.

**Comment by phansys** [comment]
I understand your concerns, but my PR is 20 months old, so I'm suspecting this will not be merged.
BTW, please take into account that what I have proposed should be compatible between major versions. Thus, IMO, the effort from that point of view should be focused on the build tool instead on the final implementation.
Therefore, if there is no feedback here, I'm seriously thinking about the possibility of publishing a fork based on the work from this repo, but including the changes I've proposed.
Anyway, thank you so much for your response.

**Comment by domibarton** [comment]
@phansys I understand your concerns, I really do. Unfortunately, I'm no longer able to support this API, as I've abandoned PHP a long time ago. Sorry about that.

I had a quick look at your PR's and they look very well engineered. You mentioned you're thinking about a fork. I'm not a big fan of forks, because I think in OSS we should focus more on collaboration than just forking - IMHO forking should always be the last resort. However, I understand why you're thinking about forking, especially after you didn't get a response within 20 months ;)

As you already implied before correctly, focusing on the build tool instead of the final implementation is much more important now. Unfortunately, I can't do that. But... you can ;) Apparently you've the knowhow (much more than I do at this state ^^) and you seem to be a nice guy. 

If you're OK with that, I'll add you as contributor and we can discuss how we should carry on with the project. I'd be really happy to give my project into the hands of somebody else who's capable of maintaining it and keeping it alive, or even improve it further :)

Let me know if you're interested.

Cheers
Domi

**Comment by toine7m** [comment]
STOP

Le ven. 28 juin 2019 à 11:12, Dominique Barton <notifications@github.com> a
écrit :

**Comment by ojgarciab** [comment]
> If you're OK with that, I'll add you as contributor and we can discuss how we should carry on with the project. I'd be really happy to give my project into the hands of somebody else who's capable of maintaining it and keeping it alive, or even improve it further :)

Say yes! say yes! @phansys

From an improvement as your pull request we can have a very good base to continue sending contributions.

In addition, if the master branch is protected by forcing the CI to finish correctly (with unit tests) you could have one or two people reviewing each PR avoiding centralizing all the work in a single person.

**Comment by phansys** [comment]
Sure @domibarton! Thank you so much for sharing your opinion and your concerns.
I'll feel more comfortable working to improve this project too, if there is no need for a fork :+1:
When we have discussed about your ideas, I think I'll be able to organize my time in order to give this project some effort.
Thank you too @ojgarciab for offering your support!

**Comment by phansys** [comment]
Hey @domibarton, how are you?
I'm in the middle of the migration from version 3.x to 4.x. For me, it would be a good opportunity to bring some time for this project.
Please, let me know if you're able to add my account as contributor in order to ease the process.
In that case, we could create a project board in order to manage the next steps.

Thank you in advance.

**Comment by seisen** [comment]
@phansys let a fork go !

**Comment by seisen** [comment]
@phansys fork ! fork !

**Comment by domibarton** [comment]
thanks @seisen for nagging me :)

@phansys you should've an invitation!

**Comment by phansys** [comment]
Thank you so much! :tada:

**Comment by domibarton** [comment]
Thank you ;) And sry for the delay!

**Comment by gustavobborges** [comment]
Hi Guys, i'm trying use hostGet (or other functions api) in my ZabixController.. but i need use parameters like 'hostid = 10084'... 
i'm trying do it:

" 
return $this->zabbix->hostGet([
   'output' => 'extend',
   'search' => array('hostid' => '10084')
 ]);
"

but it comming all data....

anyone know why parameters not working???

thanks!!! *sorry for my english.

congrats for repository and so thanks!
